src/main/java/org/folio/helper/PurchaseOrderHelper.java

```diff
@@ -191,8 +192,18 @@ public PurchaseOrderHelper(PurchaseOrderLineHelper purchaseOrderLineHelper, Comp
    * @return completable future with {@link PurchaseOrderCollection} object on success or an exception if processing fails
    */
   public Future<PurchaseOrderCollection> getPurchaseOrders(int limit, int offset, String query, RequestContext requestContext) {
-   return buildGetOrdersPath(limit, offset, query, requestContext)
-      .compose(endpoint -> purchaseOrderStorageService.getPurchaseOrders(query, limit, offset, requestContext))
+    return acquisitionsUnitsService.buildAcqUnitsCqlExprToSearchRecords(StringUtils.EMPTY, requestContext)
+      .compose(acqUnitsCqlExpr -> {
+        String finalQuery = acqUnitsCqlExpr;
+        if (StringUtils.isNotEmpty(query)) {
+          finalQuery = combineCqlExpressions("and", acqUnitsCqlExpr, query);
+        }
+        RequestEntry requestEntry = new RequestEntry(resourcesPath(PURCHASE_ORDER_STORAGE))
+          .withQuery(finalQuery)
+          .withLimit(limit)
+          .withOffset(offset);
+        return restClient.get(requestEntry, PurchaseOrderCollection.class, requestContext);
+      })
```

Comment:
We can use use the same purchaseOrderStorageService.getPurchaseOrders(query, limit, offset, requestContext) without introducing RestClient as dependency.